react/features/authentication/middleware.ts

```diff
@@ -305,3 +280,36 @@ function _handleLogin({ dispatch, getState }: IStore) {
 
     dispatch(openTokenAuthUrl(tokenAuthServiceUrl));
 }
+
+/**
+ * Handles logout challenge. Opens logout dialog and hangs up the conference.
+ *
+ * @param {Store} store - The redux store in which the specified {@code action}
+ * is being dispatched.
+ * @param {string} logoutUrl - The url for logging out.
+ * @returns {void}
+ */
+function _handleLogout({ dispatch, getState }: IStore) {
+    const state = getState();
+    const config = state['features/base/config'];
+    const logoutUrl = config.tokenLogoutUrl;
+
+    const { conference } = state['features/base/conference'];
+    const { jwt } = state['features/base/jwt'];
+
+    if (!conference) {
+        return;
+    }
+
+    if (isTokenAuthEnabled(config) && config.tokenAuthUrlAutoRedirect && jwt) {
+
+        // user is logging out remove auto redirect indication
+        dispatch(setTokenAuthUrlSuccess(false));
+
+        if (logoutUrl) {
+            dispatch(openLogoutDialog(logoutUrl));
+        }
+    }
+
+    conference.room.moderator.logout(() => dispatch(hangup(true)));
```

Comment:
@saghul Do we need to do this on both web and native?

Reply:
I think we can drop it, right @damencho ? Or won't it wipe the session id?

Reply:
As a matter of fact, I think we do. I was testing with one participant and, of course, that participant is a moderator.

Reply:
Let's skip for RN for now.